spec/rubocop/cop/lint/constant_base_spec.rb

```diff
@@ -0,0 +1,81 @@
+# frozen_string_literal: true
+
+RSpec.describe RuboCop::Cop::Lint::ConstantBase, :config do
+  context 'with prefixed constant in class' do
+    it 'registers no offense' do
+      expect_no_offenses(<<~RUBY)
+        class Foo
+          ::Bar
+        end
+      RUBY
+    end
+  end
+
+  context 'with prefixed constant in module' do
+    it 'registers no offense' do
+      expect_no_offenses(<<~RUBY)
+        module Foo
+          ::Bar
+        end
+      RUBY
+    end
+  end
+
+  context 'with prefixed constant in neither class nor module' do
+    it 'registers an offense' do
+      expect_offense(<<~RUBY)
+        ::Bar
+        ^^ Avoid meaningless `::` prefix on constant.
```

Comment:
It can be told more simply.

```suggestion
        ^^ Remove redundant `::`.
```